pages/api/teams/[teamId]/folders/manage/[folderId]/index.ts

```diff
@@ -57,10 +56,34 @@ export default async function handle(
         },
       });
 
-      if (folder?._count.documents! > 0 || folder?._count.childFolders! > 0) {
-        return res.status(401).json({
-          message: "Folder contains documents or folders. Move them first",
+      const deleteFolderWithChildren = async (folderId: string) => {
+        const children = await prisma.folder.findMany({
+          where: {
+            parentId: folderId,
+          },
+        });
+
+        for (const child of children) {
+          await deleteFolderWithChildren(child.id);
+        }
+
+        await prisma.document.deleteMany({
+          where: {
+            folderId: folderId,
+          },
         });
```

Comment:
We need to delete the actual files (versions, pages) from the file storage as well. so please check how we are deleting a single file currently and adapt this.